crypto/pkcs12/pbkdf/pbkdf.go

```diff
@@ -0,0 +1,201 @@
+// Package pbkdf implements
+//
+// Entire package adapted from https://github.com/Azure/go-pkcs12/blob/master/pbkdf.go
+// and https://github.com/Azure/go-pkcs12/blob/master/crypto.go under the MIT
+// license.  These functions implement cryptographic processes used in decoding
+// password protected pfx objects adhering to the pkcs12 standard as documented at
+//
+// https://tools.ietf.org/html/rfc7292#appendix-B.2
+//
+//
+package pbkdf
+
+import (
+	"errors"
+	"math/big"
+	"unicode/utf16"
+	"unicode/utf8"
+)
+
+// Pbkdf implements RSA Labs' Password-Based Key Derivation Function
+func Pbkdf(hash func([]byte) []byte, u, v int, salt, password []byte, r int, ID byte, size int) (key []byte) {
```

Comment:
The function name should be PBKDF. Ideally, it would be named `Key` in the style of the [PBKDF2](godoc.org/golang.org/x/crypto/pbkdf2) and [scrypt](godoc.org/golang.org/x/crypto/scrypt) packages.